python/ray/experimental/dag/class_node.py

```diff
@@ -63,57 +89,72 @@ def options(self, **options):
         return self
 
 
-class ClassMethodNode(DAGNode):
+class ClassMethodNode(ray_dag.DAGNode):
     """Represents an actor method invocation in a Ray function DAG."""
 
     def __init__(
         self,
-        actor: ClassNode,
-        prev_call: Optional["ClassMethodNode"],
-        method_name: str,
         method_args: Tuple[Any],
         method_kwargs: Dict[str, Any],
         method_options: Dict[str, Any],
+        kwargs_to_resolve: Dict[str, Any],
     ):
-        self._method_name: str = method_name
+        # TODO: (jiaodong) revisit constraints on dag INPUT before moving out
+        # of experimental folder
+        if ray_dag.INPUT in method_args and len(method_args) > 1:
+            raise ValueError(
+                "dag.INPUT cannot be used in conjunction with other args. "
+                "Please bind with dag.INPUT as only input."
+            )
+        if ray_dag.INPUT in method_kwargs.values():
```

Comment:
also check that it isn't in method_options

Reply:
we're using pickling method now that skips tracking DAGNode's `options`. I can do this in a follow-up at DAGNode level so that `options()` should contain no object of type DAGNode at all that will take care of this case.